service/src/main/java/bio/terra/pipelines/app/controller/GlobalExceptionHandler.java

```diff
@@ -56,6 +58,30 @@ public ResponseEntity<ApiErrorReport> validationExceptionHandler(Exception ex) {
     return new ResponseEntity<>(errorReport, HttpStatus.BAD_REQUEST);
   }
 
+  @ExceptionHandler({MethodArgumentNotValidException.class})
+  public ResponseEntity<ApiErrorReport> argumentNotValidExceptionHandler(
+      MethodArgumentNotValidException ex) {
+    logger.debug("MethodArgumentNotValid exception caught by global exception handler", ex);
+    // For security reasons, we generally don't want to include the user's invalid (and potentially
+    // malicious) input in the error response, which also means we don't include the full exception.
+    // Instead, we return a generic error message about input validation.
+    Map<String, String> errors = new HashMap<>();
+    ex.getBindingResult()
+        .getAllErrors()
+        .forEach(
+            error -> {
+              String fieldName = ((FieldError) error).getField();
```

Comment:
do we have to do the cast here?  surprised thats necessary if we know what the exception we're starting with is.

Reply:
we seemingly do... `.getAllErrors1 returns a List<ObjectError>, and ObjectError doesn't have a `.getField` method. let me know if I'm missing something though!

Reply:
nevermind! there's a `.getFieldErrors` method!

Reply:
woohoo